### PR TITLE
Application terms

### PR DESCRIPTION
**Pull request by pbn4**
Closes https://github.com/bloom-housing/bloom/issues/274

### CODE REVIEW

**Comment by pbn4** [comment]
@bencpeters  I am not yet sure how to connect it to user login, since the API requires access token which can be undefined. @jaredcwhite @bk3c  Is profile data and access token guaranteed to be present when submitting in normal conditions? In other words: is it required for user to be logged in to submit an application?

**Comment by pbn4** [comment]
Ready for review. I added possibility to submit an application for unauthenticated users and some tests for that.

**Comment by pbn4** [comment]
@bk3c I'm unable to make multiline strings in terms with carriage returns so I left parsing raw HTML enabled for now. Could you point me to how it's done?

**Comment by pbn4** [comment]
@slowbot Thanks for taking time. 

There is one outstanding problem with rendering data from listing that I'm trying to solve with Jared now. Application due date is fixed at the moment.

File: shared/ui-components/src/applications/Client.d.ts [hunk 1]
```diff
@@ -0,0 +1,457 @@
+import {
+  OpenAPIClient,
+  Parameters,
+  UnknownParamsObject,
+  OperationResponse,
+  AxiosRequestConfig,
+} from "openapi-client-axios"
+
+declare namespace Components {
```
Comment (bencpeters) [inline shared/ui-components/src/applications/Client.d.ts:9]: Is this file auto-generated? If not, it seems like a bunch of duplicate definitions, there must be a better way to keep this in sync with `backend/core`?

**Comment by pbn4** [comment]
> It's not totally clear to me how much of backend/core/client.ts is autogenerated - I made some style & clean-up comments but they obviously only apply to portions that are not auto generated. Let's get a little bit of documentation for developers around exactly when generating the client files (e.g. when yarn generate:client should be run), and what, if anything, needs to be done before running that command.

Command `yarn generate:client` fetches input swagger json from a running service. So you need to start it with `yarn dev` from `backend/core` and then run `yarn generate:client`. 

> It also seems like we're moving to having a bunch of duplicate type definitions here - there are the types in backend/core/client.ts, and the shared/ui-components/Client.d.ts definitions, in addition to the entities themselves and (presumably) the DTOs used to validate incoming data with the backend. If these additional definitions are totally auto-generated by a single command, then that's fine, but if there are steps required to keep them in sync, that seems less than ideal to me.

As discussed outside github:  `shared/ui-components/Client.d.ts` was a client generated by another tool that we are not going to use and should not appear in this PR, sorry for that. I removed it. :)  The only valid client is `backend/core/client.ts` and it's entirely autogenerated, so the duplication with DTOs is not a problem (actually they are a source of truth for that client because they are what NestJS/swagger takes to generate `/api-json` file).

**Pull request merged**